packages/flutter/test/widgets/heroes_test.dart

```diff
@@ -1448,4 +1448,40 @@ void main() {
     ));
     expect(find.text('two'), findsOneWidget);
   });
+
+  testWidgets('Hero within a Hero, throws', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: Material(
+          child: Hero(
+            tag: 'a',
+            child: Hero(
```

Comment:
Would be good to check this case too:
```
child: Builder(builder: (BuildContext context) => Hero(...)
```

And likewise for LayoutBuilder

Reply:
Done